src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2337,6 +2384,25 @@ export class ResultsViewPageStore {
         }
     }, []);
 
+    readonly nonSelectedMolecularProfiles = remoteData<MolecularProfile[]>({
+        await:()=>[this.molecularProfilesInStudies, this.selectedMolecularProfiles, this.molecularProfileIdToMolecularProfile],
+        invoke:() => {
+            const selectedMolecularProfileIds = this.selectedMolecularProfiles.result!.map((profile)=>profile.molecularProfileId);         
+            const allMolecularProfileIds = this.molecularProfilesInStudies.result!.map((profile)=>profile.molecularProfileId);           
+            const nonSelectedMolecularProfileIds = _.difference(allMolecularProfileIds, selectedMolecularProfileIds);
+            const nonSelectedMolecularProfiles = _.map(nonSelectedMolecularProfileIds, (id) => {return this.molecularProfileIdToMolecularProfile.result[id]})
+            return Promise.resolve(nonSelectedMolecularProfiles);
+        }
+    }, []);
+
+    // If we have same profile accros multiple studies, they should have the same name, so we can group them by name to get all related molecular profiles in multiple studies.
+    readonly nonSelectedMolecularProfilesGroupByName = remoteData<_.Dictionary<MolecularProfile[]>>({
```

Comment:
I'm not sure what _.Dictionary is.   Probably better just to return { [profileName:string]: whatever } which we use elsewhere.  It communicates more information about key.

Reply:
Ok!